From e76310ac73e4d9d15aaceb7ba3fc470e3563f3c7 Mon Sep 17 00:00:00 2001
From: Sebastien PASDELOUP <sebastien.pasdeloup-ext@st.com>
Date: Mon, 22 Apr 2024 11:52:05 +0200
Subject: [PATCH 586/841] feat(stm32mp2): remove function
 stm32_iwdg_shadow_update()

TF-A do not configure OTP to disable IWDG during standby and stop.
Delete function stm32_iwdg_shadow_update as it's not used anymore.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git abcb3016df895c693492bdd2ce9960d3fab00098 feat(stm32mp2): remove function stm32_iwdg_shadow_update()]
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/366926
Change-Id: If204383b9e814f7e60b04318bd35d1ad81dc9e80
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/372969
Tested-by: Sebastien PASDELOUP <sebastien.pasdeloup-ext@st.com>
Domain-Review: Sebastien PASDELOUP <sebastien.pasdeloup-ext@st.com>
Reviewed-by: Sebastien PASDELOUP <sebastien.pasdeloup-ext@st.com>
---
 plat/st/stm32mp2/stm32mp2_private.c | 15 ---------------
 1 file changed, 15 deletions(-)

diff --git a/plat/st/stm32mp2/stm32mp2_private.c b/plat/st/stm32mp2/stm32mp2_private.c
index 663bd31bf..cbedd7cc5 100644
--- a/plat/st/stm32mp2/stm32mp2_private.c
+++ b/plat/st/stm32mp2/stm32mp2_private.c
@@ -565,24 +565,9 @@ uint32_t stm32_iwdg_get_otp_config(uint32_t iwdg_inst)
 		iwdg_cfg |= IWDG_HW_ENABLED;
 	}
 
-	if ((otp_value & HCONF1_OTP_IWDG_FZ_STOP_MASK(iwdg_inst)) != 0U) {
-		iwdg_cfg |= IWDG_DISABLE_ON_STOP;
-	}
-
-	if ((otp_value & HCONF1_OTP_IWDG_FZ_STANDBY_MASK(iwdg_inst)) != 0U) {
-		iwdg_cfg |= IWDG_DISABLE_ON_STANDBY;
-	}
-
 	return iwdg_cfg;
 }
 
-#if defined(IMAGE_BL2)
-uint32_t stm32_iwdg_shadow_update(uint32_t iwdg_inst, uint32_t flags)
-{
-	return BSEC_OK;
-}
-#endif
-
 bool stm32mp_skip_boot_device_after_standby(void)
 {
 	return false;
-- 
2.47.2

